### PR TITLE
Create feedback page

### PR DESCRIPTION
**Pull request by andrey-p**
**Reviewer:** @MariagraziaAlastra 
**CC:** @nilnilnil @alohaas 
**Depends on:** Internal PR (coming right up!)

## Description:

This removes the mailto feedback links and adds a dedicated feedback page. Still a WIP.

TODO:

- add unit tests
- test on Safari

## Steps to test this PR:

**On the popup**

1. Open a site
2. Open the popup and click "report broken site"
3. Feedback page should open with "report broken site" mode on and the URL prefilled

**Hamburger menu**

1. Open a site
2. Open the popup and then the hamburger menu
3. Click "send feedback" -> the feedback page should open with "report broken site" mode off
4. Click "report broken site" -> the feedback page should open with "report broken site" mode on and the URL prefilled

**Options page**

1. Open the options page
2. Click "send feedback" -> the feedback page should open with "report broken site" mode off
3. Click "report broken site" -> the feedback page should open with "report broken site" mode on and the URL should NOT be prefilled

**When on the feedback form**

1. The Submit button should turn blue when all visible forms are filled
2. Checking / unchecking the box should switch "report broken site" mode on and off
3. When you submit, the thank you message should be different between the two modes

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: shared/js/ui/models/feedback-form.es6.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+FeedbackForm.prototype = window.$.extend({},
+    Parent.prototype,
+    {
+        modelName: 'feedbackForm',
+
+        submit: function () {
+            if (!this.canSubmit) { return }
+
+            console.log('SUBMIT!!!')
```
Comment (nilnilnil) [inline shared/js/ui/models/feedback-form.es6.js]: WIP?

File: shared/js/ui/pages/mixins/parse-query-string.es6.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+module.exports = {
```
Comment (nilnilnil) [inline shared/js/ui/pages/mixins/parse-query-string.es6.js]: Why is this necessary?

File: shared/js/ui/pages/options.es6.js [hunk 3]
```diff
@@ -8,8 +8,7 @@ const WhitelistModel = require('./../models/whitelist.es6.js')
 const whitelistTemplate = require('./../templates/whitelist.es6.js')
 const BackgroundMessageModel = require('./../models/background-message.es6.js')
 const parseUserAgentString = require('./../models/mixins/parse-user-agent.es6.js')
-const renderFeedbackHref = require('./../templates/shared/render-feedback-href.es6.js')
-const renderBrokenSiteHref = require('./../templates/shared/render-broken-site-href.es6.js')
+const browserUIWrapper = require('./../base/$BROWSER-ui-wrapper.es6.js')
```
Comment (nilnilnil) [inline shared/js/ui/pages/options.es6.js]: neat use of var

File: shared/js/ui/models/feedback-form.es6.js (hunk 1 above)
Comment (MariagraziaAlastra) [reply shared/js/ui/models/feedback-form.es6.js]: Yep this is WIP

File: shared/js/ui/pages/mixins/parse-query-string.es6.js (hunk 2 above)
Comment (MariagraziaAlastra) [reply shared/js/ui/pages/mixins/parse-query-string.es6.js]: It's a mixin, so it's meant to be shareable from multiple places via require()

File: shared/js/ui/views/feedback-form.es6.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            this.model.updateCanSubmit()
+        },
+
+        _onMessageChange: function () {
+            this.model.set('message', this.$message.val())
+            this.model.updateCanSubmit()
+        },
+
+        _onSubmitClick: function (e) {
+            this.model.submit()
```
Comment (MariagraziaAlastra) [inline shared/js/ui/views/feedback-form.es6.js]: Hey @andrey-p can you check if the button is disabled and in that case, return before attempting to submit?
Right now it submits regardless of whether the fields are filled in

File: shared/js/ui/pages/feedback.es6.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            this.setBrowserClassOnBodyTag()
+
+            let params = this.parseQueryString(window.location.search)
+            let browserInfo = this.parseUserAgentString()
+
+            this.form = new FeedbackFormView({
+                appendTo: window.$('.js-feedback-form'),
+                model: new FeedbackFormModel({
+                    isBrokenSite: params.broken,
+                    url: decodeURIComponent(params.url || ''),
```
Comment (MariagraziaAlastra) [inline shared/js/ui/pages/feedback.es6.js]: @andrey-p I assume sending the entire URL is intentional? In production we only send the domain.
I don't have a strong opinion, but this approach is probably better if we want to understand the exact issue, so 👍

**Comment by MariagraziaAlastra** [comment]
@nilnilnil Unit tests are listed as a TODO in the description, @andrey-p will add them before this is ready to merge. Thanks!

@andrey-p At a first glance it looks good, just left a couple of comments for really small stuff 😄

File: shared/js/ui/views/feedback-form.es6.js (hunk 4 above)
Comment (andrey-p) [reply shared/js/ui/views/feedback-form.es6.js]: Good call, thank you!

File: shared/js/ui/pages/feedback.es6.js (hunk 5 above)
Comment (andrey-p) [reply shared/js/ui/pages/feedback.es6.js]: Yeah, I think this will make our lives easier in the long run. There's currently also a field where the user could paste their URL, which (I think) most users also used. So this will make users' lives easier also :)

File: shared/js/ui/pages/options.es6.js (hunk 3 above)
Comment (andrey-p) [reply shared/js/ui/pages/options.es6.js]: The `$BROWSER` thing? Yeah, this was for the Safari port where we the extension internals are vastly different, but we could still keep a similar API between the two. It made the port much easier.

File: shared/js/ui/models/feedback-form.es6.js (hunk 1 above)
Comment (andrey-p) [reply shared/js/ui/models/feedback-form.es6.js]: Yeah, just so we can get the bulk of the review out of the way for today.

**Comment by MariagraziaAlastra** [comment]
Alright, this looks good! Thanks @andrey-p! 👍
